sources/Subs.php

```diff
@@ -1793,11 +1793,26 @@ function isValidEmail($value)
  *                 added if none is found (optional, default array('http://', 'https://'))
  * @return string - The url with the protocol
  */
-function addProtocol($url, $protocols = array('http://', 'https://'))
+function addProtocol($url, $protocols = array())
 {
-	foreach ($protocols as $protocol)
+	if (empty($protocols))
+	{
+		$pattern = '~^(http://|https://)~i';
```

Comment:
Nothing to change, but just wanted to note how this vs https?:// works ... if you feed each with https://asfdasfasd whats there will take 15 steps to find the "hit" since it has to fail and then backtrace to the start.  The one I just typed, should only take 9 steps as it will not require a backtrace.   Either would take 9 steps if fed http.